### ISSUE TITLE
Allow free shipping discount to target particular shipping method

### ISSUE THREAD

**Issue by clifgriffin** [ingenesis/shopp#3431]
A few things here:
1. Currently free shipping removes all shipping options and forces a free method. 
2. In addition to allowing customers to target a particular shipping method (not module, an actual method a live rates provider might return), it would be ideal to provide an option or an easy way to subtract the free option total from the remaining shipping options as this is a very common use case for clients historically.

**Comment by sierrab123** [comment]
I think that your analysis may be off here (!!!!!!) When you do not select shipping in the product editor, it ensures that that product counts as zero in any calculation.

It does not require free shipping to be enabled, all that it requires is any shipping methods that you have in place have a zero option in the rates e.g. Rate one would be zero and above, rate 2, one and above.

Not sure how this plays out with live rates, but it is a work around when using local modules.

**Comment by Just-m** [comment]
I guess what Clif is getting at is:
Standard shipping (1wk) $13
Fast shipping (3 days) $25
Super shipping (today) $50

On freeshipping orders, the options are removed and free shipping is put in place. The merchant will most likely use the standard shipping option to send the products.
What if the customer wants it today? Clif would like to make it possible for the customer to select Super shipping at $37

**Comment by clifgriffin** [comment]
@Just-m Yes, exactly. That would be the stretch goal.  The realistic goal to start would be just zeroing out the standard shipping.

**Comment by jonathandavis** [comment]
FWIW I like this idea. :)

**Comment by jonathandavis** [comment]
Free shipping is a terribly difficult problem. The way we have it implemented is the ​*easiest*​ scenario. If Free Shipping is triggered, we give them a free shipping option and leave it to the merchant to decide what shipping method they want to use.

I think the expectation is that merchants still want to offer a couple of ways to ship but allow them to be free. 

The mixed-cart case adds to the complexity. What if they have a product that qualifies for free shipping and one that most certainly does not and can't. A book/DVD that qualifies for free shipping, but a bed frame that is too large and bulky.

So maybe the solution here is to add a "Free Shipping" option, but leave all the paid options including standard/lowest-cost even if it is the one that may be used by merchants.

**Comment by clifgriffin** [comment]
@jonathandavis I don't think we try to solve the mixed cart issue. The merchant can limit that through cart item count conditions I think that should allow them to determine if the cart is mixed. 

I think good enough for now is just adding support, even if it isn't workable in every scenario.
